skcosmo/sample_selection/_base.py

```diff
@@ -346,3 +351,134 @@ def __init__(
             full=full,
             random_state=random_state,
         )
+
+
+class GCH:
+    """
+    Transformer that performs Greedy Sample Selection by constructing a
```

Comment:
Going through and making a few comments. Comment 1: GCH is not a greedy selector (see https://en.wikipedia.org/wiki/Greedy_algorithm)